test/services/user.test.js

```diff
@@ -70,4 +70,66 @@ describe('\'User\' service', () => {
     assert.include(result.achievements[0], {name: 'TestAchievement'});
     assert.notInclude(result.achievements[0], {name: 'HiddenAchievement'});
   });
+
+
+  it('returns the correct level when using manual levels', async () => {
+    await app.service('xp').create({
+      name: 'XP',
+      user_id: user_id,
+      amount: 300
+    });
+
+    const result = await app.service('user').get(user_id);
+    assert.deepEqual(result.user_id, user_id);
+    assert.deepEqual(result.level, 3);
+  });
+
+  it('returns the correct level when using linear levels', async () => {
+    app.set('rules', require('../../src/rule-parser')(__dirname + '/../config/linear_level_config.yml'));
+
+    await app.service('xp').create({
+      name: 'XP',
+      user_id: user_id,
+      amount: 650
+    });
+
+    const result = await app.service('user').get(user_id);
+    assert.deepEqual(result.user_id, user_id);
+    assert.deepEqual(result.level, 4);
+  });
+
+  it('returns the correct level when using exponential levels', async () => {
+    app.set('rules', require('../../src/rule-parser')(__dirname + '/../config/exponential_level_config.yml'));
+
+    await app.service('xp').create({
+      name: 'XP',
+      user_id: user_id,
+      amount: 650
+    });
+
+    const result = await app.service('user').get(user_id);
+    assert.deepEqual(result.user_id, user_id);
+    assert.deepEqual(result.level, 2);
+  });
+
+  it('throws the correct error when given an invalid type', async () => {
+    app.set('rules', require('../../src/rule-parser')(__dirname + '/../config/invalid_level_config.yml'));
+
+    await app.service('xp').create({
+      name: 'XP',
+      user_id: user_id,
+      amount: 50
+    });
+
+    let hadError = false;
+
+    try {
+      await app.service('user').get(user_id);
+    } catch (error) {
+      hadError = true;
+      assert.deepEqual(error.message, 'Levels Type should be one of "manual", "linear" or "exponential"');
+    }
+
+    assert.isTrue(hadError);
```

Comment:
This could be simplified to 
```js
    try {
      await app.service('user').get(user_id);
      assert.fail();
    } catch (error) {
      assert.deepEqual(error.message, 'Levels Type should be one of "manual", "linear" or "exponential"');
    }
```